code/modules/mob/living/simple_animal/hostile/gorilla/gorilla.dm

```diff
@@ -38,75 +38,144 @@
-				var/obj/item/bodypart/BP = X
-				if(BP.body_part != HEAD && BP.body_part != CHEST)
-					if(BP.dismemberable)
-						parts += BP
-			return parts
-
-/mob/living/simple_animal/hostile/gorilla/AttackingTarget()
+	for(var/obj/item/bodypart/part as anything in carbon_target.bodyparts)
+		if(part.body_part == HEAD || part.body_part == CHEST)
+			continue
+		if(part.dismemberable)
+			continue
+		parts += part
+	return parts
+
+/mob/living/simple_animal/hostile/gorilla/AttackingTarget(atom/attacked_target)
+	. = ..()
+	if(!.)
+		return
+
 	if(client)
 		oogaooga()
-	var/list/parts = target_bodyparts(target)
-	if(parts)
-		if(!parts.len)
-			return FALSE
-		var/obj/item/bodypart/BP = pick(parts)
-		BP.dismember()
-		return ..()
-	. = ..()
-	if(. && isliving(target))
-		var/mob/living/L = target
+
+	var/list/parts = get_target_bodyparts(target)
+	if(length(parts))
+		var/obj/item/bodypart/to_dismember = pick(parts)
+		to_dismember.dismember()
+		return
+
+	if(isliving(target))
+		var/mob/living/living_target = target
 		if(prob(80))
-			var/atom/throw_target = get_edge_target_turf(L, dir)
-			L.throw_at(throw_target, rand(1,2), 7, src)
+			living_target.throw_at(get_edge_target_turf(living_target, dir), rand(1, 2), 7, src)
+
 		else
-			L.Paralyze(20)
-			visible_message(span_danger("[src] knocks [L] down!"))
+			living_target.Paralyze(2 SECONDS)
+			visible_message(span_danger("[src] knocks [living_target] down!"))
 
 /mob/living/simple_animal/hostile/gorilla/CanAttack(atom/the_target)
-	var/list/parts = target_bodyparts(target)
-	return ..() && !ismonkey(the_target) && (!parts  || parts.len > 3)
-
+	var/list/parts = get_target_bodyparts(target)
+	return ..() && !ismonkey(the_target) && (!parts || length(parts) > 3)
 
 /mob/living/simple_animal/hostile/gorilla/CanSmashTurfs(turf/T)
 	return iswallturf(T)
 
-
 /mob/living/simple_animal/hostile/gorilla/gib(no_brain)
 	if(!no_brain)
-		var/mob/living/brain/B = new(drop_location())
-		B.name = real_name
-		B.real_name = real_name
-		if(mind)
-			mind.transfer_to(B)
-	..()
+		var/mob/living/brain/gorilla_brain = new(drop_location())
+		gorilla_brain.name = real_name
+		gorilla_brain.real_name = real_name
+		mind?.transfer_to(gorilla_brain)
+	return ..()
 
 /mob/living/simple_animal/hostile/gorilla/handle_automated_speech(override)
 	if(speak_chance && (override || prob(speak_chance)))
 		playsound(src, 'sound/creatures/gorilla.ogg', 50)
-	..()
+	return ..()
 
 /mob/living/simple_animal/hostile/gorilla/can_use_guns(obj/item/G)
 	to_chat(src, span_warning("Your meaty finger is much too large for the trigger guard!"))
 	return FALSE
 
-
 /mob/living/simple_animal/hostile/gorilla/proc/oogaooga()
 	oogas++
 	if(oogas >= rand(2,6))
 		playsound(src, 'sound/creatures/gorilla.ogg', 50)
 		oogas = 0
+
+/mob/living/simple_animal/hostile/gorilla/cargo_domestic
+	name = "Cargorilla" // Overriden, normally
+	desc = "Cargo's pet gorilla."
+	maxHealth = 200
+	health = 200
+	faction = list(FACTION_STATION)
```

Comment:
What effects does this have on who they target? HoS’ pets are Neutral

Reply:
their AI won't target any players (crew jobs start with `FACTION_STATION`), (even though they can't attack)